components/gitpod-cli/pkg/utils/logError.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package utils
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"google.golang.org/grpc"
+	"net/http"
+	"net/url"
+	"os"
+
+	gitpod "github.com/gitpod-io/gitpod/gitpod-cli/pkg/gitpod"
+	supervisor_helper "github.com/gitpod-io/gitpod/gitpod-cli/pkg/supervisor-helper"
+	ide_metrics "github.com/gitpod-io/gitpod/ide-metrics-api"
+	"github.com/go-errors/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+func LogError(ctx context.Context, errToReport error, errorMessage string) {
+	log.WithError(errToReport).Error(errorMessage)
+
+	file, err := os.OpenFile(os.TempDir()+"/gitpod-cli-errors.log", os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0666)
+	if err == nil {
+		log.SetOutput(file)
```

Comment:
I'm not sure we should do it cc @iQQBot could you recommend?

Reply:
ℹ️ We've followed this pattern from other packages' logs in gitpod repo:

<img width="965" alt="image" src="https://user-images.githubusercontent.com/418083/195652420-4629785c-68c2-455f-8dfe-808ab755140c.png">

Reply:
But this file wasn't supposed to be submitted.

The errors are actually sent to `ide-metrics-api` when they happen, via a POST request.

https://github.com/gitpod-io/gitpod/blob/4344ff0517c1453a940b8fcb9db1df9ebf805565/components/gitpod-cli/pkg/utils/logError.go#L69-L75

In the last mob session, we just discussed those errors should be written to some file, but we didn't discuss where. Please add [here](https://github.com/gitpod-io/gitpod/issues/13847) a reminder to change the file location in the follow-up PR.